Tests/GauntletTests/Assertions/EqualWithAccuracyAssertsTests.swift

```diff
@@ -0,0 +1,179 @@
+//
+//  EqualWithAccuracyAssertsTests.swift
+//
+//  MIT License
+//
+//  Copyright (c) [2023] The Kroger Co. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in all
+//  copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//  SOFTWARE.
+
+import Foundation
+import Gauntlet
+import XCTest
+
+class EqualWithAccuracyAssertsTestCase: XCTestCase {
+
+    // MARK: - Equal
+
+    /// Tests the basic success behavior using Assertion tests.
+    func testIsEqualWithAccuracySuccess() {
+        // Given
+        let expectedLine = 321
+
+        // When
+        let equalAssertion = TestAssertion(on: 5).isEqualTo(5, accuracy: 1, line: expectedLine)
+        let withAccuracyAssertion = TestAssertion(on: 5).isEqualTo(6, accuracy: 1, line: expectedLine)
+
+        // Then
+        Assert(that: equalAssertion).isSuccess(expectedName: "isEqualTo(, accuracy)", expectedLine: expectedLine)
+        Assert(that: withAccuracyAssertion).isSuccess(expectedName: "isEqualTo(, accuracy)", expectedLine: expectedLine)
+    }
+
+    /// Tests the basic failure behavior using Assertion tests.
+    func testIsEqualWithAccuracyFailure() {
+        // Given
+        let expectedLine = 123
+
+        // When
+        let assertion = TestAssertion(on: 4).isEqualTo(1, accuracy: 2, line: expectedLine)
```

Comment:
`TestAssertion(on:)` naming confused me at first glance. I'm not sure what else would work, but it wasn't immediately clear that it was related in some way to `Assert(that:)`

Reply:
Updated this to `TestAnAssertion(on: )` which I think it reads a bit better. I agree that this can be confusing, especially seeing it the first time. Testing custom assertion operators is something I'll be writing documentation for, and hope to further clarify there. This is a pattern that's only used when testing custom assertion operators, not in normal tests someone would write. But I do want to make sure things are named clearly enough that no one accidentally uses this function for a normal test.